.github/workflows/ragengine-e2e-workflow.yml

```diff
@@ -0,0 +1,120 @@
+name: ragengine-e2e-workflow
+
+on:
+  workflow_call:
+    inputs:
+      git_sha:
+        type: string
+        required: true
+      node_provisioner:
+        type: string
+        required: false
+        default: gpuprovisioner
+      tag:
+        type: string
+      isRelease:
+        type: boolean
+        default: false
+      registry:
+        type: string
+      region:
+        type: string
+        description: "the azure location to run the e2e test in"
+        default: "eastus"
+      k8s_version:
+        type: string
+        default: "1.30.0"
+
+jobs:
+  base-setup:
+    uses: ./.github/workflows/e2e-base-setup.yaml
+    with:
+      git_sha: ${{ inputs.git_sha }}
+      node_provisioner: ${{ inputs.node_provisioner }}
+      tag: ${{ inputs.tag }}
+      isRelease: ${{ inputs.isRelease }}
+      registry: ${{ inputs.registry }}
+      region: ${{ inputs.region }}
+      k8s_version: ${{ inputs.k8s_version }}
+
+  e2e-tests:
+    runs-on: [ "self-hosted", "hostname:kaito-e2e-github-runner" ]
+    name: e2e-tests-${{ inputs.node_provisioner }}
+    needs: base-setup
+    permissions:
+      contents: read
+      id-token: write # This is required for requesting the JWT
+    
+    env:
+      AZURE_CLUSTER_NAME: ${{ needs.base-setup.outputs.cluster_name }}
+      AZURE_RESOURCE_GROUP: ${{ needs.base-setup.outputs.cluster_name }}
+      TEST_SUITE: ${{ inputs.node_provisioner }}
+      E2E_ACR_REGISTRY: ${{ needs.base-setup.outputs.cluster_name }}.azurecr.io
+      E2E_ACR_REGISTRY_SECRET: ${{ needs.base-setup.outputs.cluster_name }}-acr-secret
+      GO_VERSION: "1.23"
+      KARPENTER_NAMESPACE: "karpenter"
+      GPU_PROVISIONER_NAMESPACE: "gpu-provisioner"
+
+    steps:
+      - name: Set Registry
+        if: ${{ inputs.isRelease }}
+        run: |
+          echo "REGISTRY=${{ inputs.registry }}" >> $GITHUB_ENV
+          echo "VERSION=$(echo ${{ inputs.tag }} | tr -d v)" >> $GITHUB_ENV
+      
+      - name: build kaito RAG Engine image
+        if: ${{ !inputs.isRelease }}
+        shell: bash
+        run: |
+          make docker-build-ragengine
+        env:
+          REGISTRY: ${{ env.E2E_ACR_REGISTRY }}
+      
+      - name: Remove existing Go modules directory
+        run: sudo rm -rf ~/go/pkg/mod
+      
+      - name: Set up Go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v5.2.0
+        with:
+          go-version: ${{ env.GO_VERSION  }}
+
+      - name: Install Helm
+        uses: azure/setup-helm@v4
+      
+      - name: Install KAITO RAG Engine helm chart
+        shell: bash
+        run: |
+          make az-patch-install-ragengine-helm
+          kubectl wait --for=condition=available deploy "kaito-ragengine" -n kaito-ragengine --timeout=300s
+        env:
+          REGISTRY: ${{ env.E2E_ACR_REGISTRY }}
+
+      - name: Log ${{ inputs.node_provisioner }}
```

Comment:
Why do we want to log those pods - gpu-provisioner, kaito-workspace, rag-engine?